src/LanguageServer/Protocol/Handler/SourceGenerators/SourceGeneratedDocumentText.cs

```diff
@@ -1,9 +1,11 @@
-// Licensed to the .NET Foundation under one or more agreements.
+﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
 using System.Text.Json.Serialization;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
 
-internal sealed record SourceGeneratedDocumentText([property: JsonPropertyName("text")] string? Text);
\ No newline at end of file
+internal sealed record SourceGeneratedDocumentText(
+    [property: JsonPropertyName("resultId")] string? ResultId,
+    [property: JsonPropertyName("text")] string? Text);
```

Comment:
nit: consider doc'ing what all 4 permutations of this mean (with null or non null).